### PR TITLE
solution for hm

### PR DESCRIPTION
**Pull request by MaksymMatuliak**

### CODE REVIEW

**Comment by Tarabarnik** [comment]
please, check yourself with the checklist first and then send to review

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+        this.colour = colour;
+        this.model = model;
+        this.capacity = capacity;
+        this.maxSpeed = maxSpeed;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        Car cloned = new Car(engine, colour, model, capacity, maxSpeed);
+        return cloned;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:21]:
```suggestion
        return new Car(engine.clone(), colour, model, capacity, maxSpeed);
```

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+        this.volume = volume;
+        this.model = model;
+        this.typeOfFuel = typeOfFuel;
+        this.diameter = diameter;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return this;
+    }
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java:21]: do you think it's right?

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+        this.colour = colour;
+        this.model = model;
+        this.capacity = capacity;
+        this.maxSpeed = maxSpeed;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        Car cloned = new Car(engine.clone(), colour, model, capacity, maxSpeed);
+        return cloned;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java:21]: don't create variable, just return the instance

File: src/main/java/core/basesyntax/Engine.java [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+        this.power = power;
+        this.volume = volume;
+        this.model = model;
+        this.typeOfFuel = typeOfFuel;
+        this.diameter = diameter;
+    }
+
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
+        return this;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java:20]: why do you return `this`?

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/Car.java]: done

File: src/main/java/core/basesyntax/Car.java (hunk 3 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/Car.java:21]: done

File: src/main/java/core/basesyntax/Engine.java (hunk 4 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/Engine.java:20]: done

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (MaksymMatuliak) [reply src/main/java/core/basesyntax/Engine.java]: done

**Pull request closed without merging**